MiniDSP.h

```diff
@@ -0,0 +1,157 @@
+/* Copyright (C) 2014 Kristian Lauszus, TKJ Electronics. All rights reserved.
+ This software may be distributed and modified under the terms of the GNU
+ General Public License version 2 (GPL2) as published by the Free Software
+ Foundation and appearing in the file GPL2.TXT included in the packaging of
+ this file. Please note that GPL2 Section 2[b] requires that all works based
+ on this software must also be made publicly available under the terms of
+ the GPL2 ("Copyleft").
+ Contact information
+ -------------------
+ Kristian Lauszus, TKJ Electronics
+ Web      :  http://www.tkjelectronics.com
+ e-mail   :  kristianl@tkjelectronics.com
+ */
+
+#pragma once
+
+#include "controllerEnums.h"
+#include "hiduniversal.h"
+
+#define MINIDSP_VID 0x2752 // MiniDSP
+#define MINIDSP_PID 0x0011 // MiniDSP 2x4HD
+
+/**
+ * Arduino MiniDSP 2x4HD USB Host Driver by Dennis Frett.
+ *
+ * This class implements support for the MiniDSP 2x4HD via USB.
+ * Based on NodeJS implementation by Mathieu Rene:
+ * https://github.com/mrene/node-minidsp and the Python implementation by Mark
+ * Kubiak: https://github.com/markubiak/python3-minidsp.
+ *
+ * It uses the HIDUniversal class for all the USB communication.
+ */
+class MiniDSP : public HIDUniversal {
+public:
+  /**
+   * Constructor for the MiniDSP class.
+   * @param  p   Pointer to the USB class instance.
+   */
+  MiniDSP(USB *p) : HIDUniversal(p){};
+
+  /**
+   * Used to check if a MiniDSP 2x4HD is connected.
+   * @return Returns true if it is connected.
+   */
+  bool connected() {
+    return HIDUniversal::isReady() && HIDUniversal::VID == MINIDSP_VID &&
+           HIDUniversal::PID == MINIDSP_PID;
+  };
+
+  /**
+   * Used to call your own function when the device is successfully initialized.
+   * @param func Function to call.
+   */
+  void SetOnInitCallback(void (*func)(void)) { onInitCallback = func; };
```

Comment:
Please rename to `attachOnInit`, so it is the same as all the other drivers fx: https://github.com/felis/USB_Host_Shield_2.0/blob/master/PS5USB.h#L53-L59